server/apps/users/views.py

```diff
@@ -131,3 +124,94 @@ def get(request):
+    def put(request, **kwargs):
+        user = request.user
+        todo_id = request.data['todo_id']
+
+        # Get the to'do that user want to update from database and check if to'do in our database has current user as a editor
+        todo = Todo.objects.filter(id=todo_id).first()
+        editors = todo.editors.all()
+
+        # If current user is not an owner or editor for the to'do then don't process the request
+        if user != todo.owner and user not in editors:
+            return Response(OPERATION_NOT_ALLOWED)
+
+        # Set current user as updated by
+        request.data['updated_by'] = user.name
+
+        serializer = TodoSerializer(todo, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response(serializer.data)
+
+    # Delete to'do
+    @staticmethod
+    def delete(request, **kwargs):
+        todo_id: int = request.GET['todo-id']
+        user = request.user
+
+        # Get the to'do that user want to update from database and check if to'do in our database has current user as a editor
+        todo = Todo.objects.filter(id=todo_id).first()
+        editors = todo.editors.all()
+
+        # If current user is not an owner or editor for the to'do then don't process the request
+        if user != todo.owner and user not in editors:
+            return Response(OPERATION_NOT_ALLOWED)
+
+        todo.delete()
+
+        return Response(TODO_REMOVED)
+
+
+class TodoUtils:
+    @staticmethod
+    def list_of_todos(request) -> list:
+        todos = TodoView.get(request)
+        todos = json.dumps(todos.data)
+        todos = json.loads(todos)
+        return todos
+
+
+class SearchTodo(generics.RetrieveAPIView, TodoUtils):
+    @classmethod
+    def get(cls, request, **kwargs):
+        # Pass title as a query parameter in the request URL
+        title: str = request.GET.get('title')
+        todos = cls.list_of_todos(request)
+        todos = [
+            todo
+            for todo in todos if todo['title'] == title
+        ]
+
+        return Response(todos)
+
+
+class SortTodo(generics.ListAPIView, TodoUtils):
+    @classmethod
+    def get(cls, request, **kwargs):
+        # Pass sort_type as a query parameter in the request URL
+        sort_type: str = request.GET.get('sort_type').lower()
+        if sort_type == "id":
+            todos = cls.sort_by_id(request)
+        elif sort_type == "date":
+            todos = cls.sort_by_date(request)
+        else:
+            return Response(OPERATION_NOT_FOUND_ERROR)
+
+        return Response(todos)
+
+    @classmethod
+    def sort_by_id(cls, request):
+        todos = cls.list_of_todos(request)
+        result = sorted(todos, key=lambda todo: todo['id'], reverse=True)
+        return result
+
+    @classmethod
+    def sort_by_date(cls, request):
+        todos = cls.list_of_todos(request)
+        result = sorted(todos, key=lambda todo: todo['date'], reverse=True)
+        return result
```

Comment:
Refer this https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter

Reply:
Updated